lightning/src/ln/channel.rs

```diff
@@ -5626,11 +5642,19 @@ impl<SP: Deref> Channel<SP> where
 		};
 
 		// From here on out, we may not fail!
+		let shutdown_result;
+		let unbroadcasted_batch_funding_txid = self.context.unbroadcasted_batch_funding_txid();
 		self.context.target_closing_feerate_sats_per_kw = target_feerate_sats_per_kw;
 		if self.context.channel_state & !STATE_FLAGS < ChannelState::FundingSent as u32 {
```

Comment:
Instead of defining `shutdown_result` on L5645 won't it be better to have this if/else expression return a `ShutdownResult` and then capture it in the `shutdown_result` variable.

Reply:
Done, capturing `shutdown_result` from if-else expressions in a couple places now.